src/encoded/ingestion_listener.py

```diff
@@ -0,0 +1,670 @@
+    with vapp_for_email(email=email, app=app) as vapp:
+        if DEBUG_SUBMISSIONS:
+            PRINT("PROCESSING FOR %s" % email)
+        submission = SubmissionFolio(vapp=vapp, ingestion_type=ingestion_type, submission_id=submission_id, log=None)
+        handler = get_ingestion_processor(ingestion_type)
+        result = handler(submission)
+        if DEBUG_SUBMISSIONS:
+            PRINT("DONE PROCESSING FOR %s" % email)
+        return {
+            "result": result,
+            "ingestion_type": ingestion_type,
+            "submission_id": submission_id,
+        }
+
+
+def verify_vcf_file_status_is_not_ingested(request, uuid, *, expected=True):
+    """ Verifies the given VCF file has not already been ingested by checking
+        'file_ingestion_status'
+    """
+    kwargs = {
+        'environ': request.environ,
+        'method': 'GET',
+        'content_type': 'application/json'
+    }
+    subreq = Request.blank('/' + uuid, **kwargs)
+    resp = request.invoke_subrequest(subreq)
+    if isinstance(resp, HTTPMovedPermanently):  # if we hit a redirect, follow it
+        subreq = Request.blank(resp.location, **kwargs)
+        resp = request.invoke_subrequest(subreq)
+    log.info('VCF File Meta: %s' % resp.json)
+    verified = bool(expected) is (resp.json.get('file_ingestion_status', None) != STATUS_INGESTED)
+    # if not verified:
+    #     import pdb; pdb.set_trace()
+    return verified
+
+
+def patch_vcf_file_status(request, uuids):
+    """ Patches VCF File status to 'Queued'
+        NOTE: This process makes queue_ingestion not scale terribly well.
+              Batching above a certain number may result in 504. There are
+              also permissions concerns here that are not dealt with.
+    """
+    for uuid in uuids:
+        kwargs = {
+            'environ': request.environ,
+            'method': 'PATCH',
+            'content_type': 'application/json',
+            'POST': json.dumps({
+                'file_ingestion_status': STATUS_QUEUED
+            }).encode('utf-8')
+        }
+        subreq = Request.blank('/' + uuid, **kwargs)
+        resp = None
+        try:
+            if verify_vcf_file_status_is_not_ingested(request, uuid):
+                resp = request.invoke_subrequest(subreq)
+        except HTTPNotFound:
+            log.error('Tried to patch %s but item does not exist: %s' % (uuid, resp))
+
+
+@view_config(route_name='queue_ingestion', request_method='POST', permission='index')
+@debug_log
+def queue_ingestion(context, request):
+    """ Queues uuids as part of the request body for ingestion. Can batch as many as desired in a
+        single request. Note that you can also pass ingestion_type, which will apply to all uuids queued.
+        The default is (SNV) vcf.
+    """
+    ignored(context)
+    uuids = request.json.get('uuids', [])
+    ingestion_type = request.json.get('ingestion_type', 'vcf')  # note that this applies to all uuids
```

Comment:
Should default to one that isn't `vcf` probably.

Reply:
Default to `ontology` then right?

Reply:
I disagree here. It shouldn't be optional and is only optional because some vcf didn't use it, so vcf is the right default.

If you want to change it to required, that's OK with me, but only after verifying that all external callers are updated. I don't like the idea of saying "well, in Fourfront there are no such callers" because then the code has to be uselessly different between Fourfront and CGAP.

Reply:
Bottom line: I would leave it defaulting to vcf. Definitely _not_ to ontology. If not vcf, the only rational other thing is to just omit the default (make it be `None`) and then raise an informative error if that situation happens.